api/audit/tasks.py

```diff
@@ -89,7 +89,9 @@ def create_audit_log_from_historical_record(
     if not (history_user or override_author or history_instance.master_api_key):
         return
 
-    environment, project = instance.get_environment_and_project()
+    environment = project = None
+    if instance.__class__.objects.filter(pk=instance.pk).exists():
+        environment, project = instance.get_environment_and_project()
```

Comment:
Thanks for the contribution, @uinstinct.

Two thoughts on the PR:

1. Let's add a unit test that reproduces the bug/problem we're trying to fix here.
1. An `AuditLog` instance lacking both `project` and `environment` relations is not valid. Even for a soft-deleted instance it shouldn't be a problem to retrieve them. In fact, I believe the actual problem is that the `Segment` model class does not implement `_get_environment` and `_get_project` methods as required by logic inside `get_environment_and_project`. This should be clearer when we have a test with the actual reproducing code.

Reply:
on it

Reply:
@uinstinct , just checking in to see when you think you'll have time to make the suggested changes?

Reply:
@uinstinct ?